File: Java/NetBeans/QLTV/src/qltv/TapChi.java
package qltv;

public class TapChi extends TaiLieu{
    protected String chuyenNganh;
    protected int so;
    protected int nam;
    
    public TapChi(String maTL,
                String tenTL,
                int viTri,
                String chuyenNganh,
                int so,
                int nam) {
        super(maTL, tenTL, viTri);
        this.chuyenNganh = chuyenNganh;
        this.so = so;
        this.nam = nam;
    }
    
    public void hienThi() {
        super.hienThi();
        System.out.println("Chuyen nganh " + chuyenNganh);
        System.out.println("So: " + so);
        System.out.println("Nam: " + nam);
    }
}

File: C#/QLThuVien/QLThuVien/TapChi.cs
namespace QLThuVien
{
    class TapChi : TaiLieu
    {
        protected string chuyenNganh;
        protected int so;
        protected int nam;

        public string ChuyenNganh { get => chuyenNganh; set => chuyenNganh = value; }
        public int So { get => so; set => so = value; }
        public int Nam { get => nam; set => nam = value; }

        public TapChi(string maTL, string tenTL, int viTri, string chuyenNganh, int so, int nam) : base(maTL, tenTL, viTri)
        {
            this.chuyenNganh = chuyenNganh;
            this.so = so;
            this.nam = nam;
        }

        public override void hienThi()
        {
            base.hienThi();
            Console.WriteLine($"Chuyen nganh: {chuyenNganh}");
            Console.WriteLine($"So: {so}");
            Console.WriteLine($"Nam: {nam}");
        }

        public override string loaiTL()
        {
            return "TapChi";
        }
    }
}